internal/tiltfile/extension/extension.go

```diff
@@ -0,0 +1,66 @@
+// Package extension implements Tilt extensions.
+// This is not the internal Starkit abstraction, but the user-visible feature.
+// In a Tiltfile, you can write `load("ext://foo", "bar")` to load the function bar
+// from the extension foo.
+package extension
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"go.starlark.net/starlark"
+
+	"github.com/windmilleng/tilt/internal/tiltfile/starkit"
+)
+
+type Extension struct {
+	ctx     context.Context
+	fetcher Fetcher
+	store   Store
+}
+
+func NewExtension(ctx context.Context, fetcher Fetcher, store Store) *Extension {
+	return &Extension{
+		ctx:     ctx,
+		fetcher: fetcher,
+		store:   store,
+	}
+}
+
+type Fetcher interface {
+	Fetch(ctx context.Context, moduleName string) (ModuleContents, error)
+}
+
+func (e *Extension) OnStart(env *starkit.Environment) error {
+	env.AddLoadInterceptor(e)
+	return nil
+}
+
+const extensionPrefix = "ext://"
+
+func (e *Extension) LocalPath(t *starlark.Thread, arg string) (string, error) {
+	if !strings.HasPrefix(arg, extensionPrefix) {
+		return "", nil
+	}
+
+	moduleName := strings.TrimPrefix(arg, extensionPrefix)
+	// If the module can't be found we fetch it below
+	localPath, err := e.store.ModulePath(e.ctx, moduleName)
```

Comment:
i'm a little bit worried about the contract around `e.ctx` here. It's not clear if this is a global context or a thread-specific context, if it has a logger attached, etc.

would it make sense to attach a context to the starlark thread?

Reply:
Hm, yeah I think that would make more sense.